From aa241e6602bf53aa438b2e3ccbf5ac3c3aa08571 Mon Sep 17 00:00:00 2001
From: Xulin Sun <xulin.sun@windriver.com>
Date: Fri, 1 Feb 2019 16:41:41 +0800
Subject: [PATCH 1/4] crypto: caam - fix 't_alg' used uninitialized in
 caam_algapi_init()

The local variable "t_alg" should be initialized before using, move to
assignment value back. In order to avoid below build warning:

drivers/crypto/caam/caamalg.c:3705:19: warning: 't_alg' may be used
uninitialized in this function [-Wmaybe-uninitialized]
       (t_alg->caam.class1_alg_type & OP_ALG_AAI_MASK) ==
               ~~~~~~~~~~~^~~~~~~~~~~~~~~~

Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/crypto/caam/caamalg.c |   14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/drivers/crypto/caam/caamalg.c b/drivers/crypto/caam/caamalg.c
index 7712447..aba7d81 100644
--- a/drivers/crypto/caam/caamalg.c
+++ b/drivers/crypto/caam/caamalg.c
@@ -3697,6 +3697,13 @@ static int __init caam_algapi_init(void)
 		if (!aes_inst && (alg_sel == OP_ALG_ALGSEL_AES))
 				continue;
 
+		t_alg = caam_alg_alloc(alg);
+		if (IS_ERR(t_alg)) {
+			err = PTR_ERR(t_alg);
+			pr_warn("%s alg allocation failed\n", alg->driver_name);
+			continue;
+		}
+
 		/*
 		 * Check support for AES modes not available
 		 * on LP devices.
@@ -3706,13 +3713,6 @@ static int __init caam_algapi_init(void)
 		    OP_ALG_AAI_XTS)
 			continue;
 
-		t_alg = caam_alg_alloc(alg);
-		if (IS_ERR(t_alg)) {
-			err = PTR_ERR(t_alg);
-			pr_warn("%s alg allocation failed\n", alg->driver_name);
-			continue;
-		}
-
 		err = crypto_register_alg(&t_alg->crypto_alg);
 		if (err) {
 			pr_warn("%s alg registration failed\n",
-- 
1.7.9.5

